enginetest/join_queries.go

```diff
@@ -236,4 +236,24 @@ var JoinQueryTests = []QueryTest{
 			{2, "second row", "third", 1},
 		},
 	},
+	{
```

Comment:
When we add a `QueryTest`, we usually want to also add the corresponding `query_plan.go` `PlanTest`. The `TestWriteQueryPlans` can help generate expected outputs if you add query stubs.